azext_iot/common/deps.py

```diff
@@ -12,11 +12,19 @@
 from azext_iot.common.config import get_uamqp_ext_version, update_uamqp_ext_version
 from azext_iot.common.pip import install
 from azext_iot.common._homebrew_patch import HomebrewPipPatch
+from azure.cli.core._profile import Profile
+from azure.cli.core.mock import DummyCli
+from azext_iot.tests.settings import UserTypes
 
 
 def ensure_uamqp(config, yes=False, repair=False):
     if get_uamqp_ext_version(config) != EVENT_LIB[1] or repair or not test_import(EVENT_LIB[0]):
-        if not yes:
+        profile = Profile(cli_ctx=DummyCli())
```

Comment:
Please don't use `DummyCli` in non-test code.  Also why is this needed, providing a `-y` to a uamqp command should skip the interaction flow?